### DISCUSSION TITLE
Does "hopp test" support passing variables between tests with the pw.env.set ?

### DISCUSSION THREAD

**Discussion by m0un10** [hoppscotch/hoppscotch#3616] [Q&A] [answered]
This works from the UI, but it didn't work when I ran from the CLI.
I ran it with `hopp test collection.json -e environment.json` and it worked up until the point that it needed to use the previously set variable.

**Comment by jamesgeorge007** [comment]
Hi, could you confirm if the discrepancy persists in the latest version of the CLI?

**Comment by jamesgeorge007** [comment, marked as answer]
This is supported in the current CLI. Values written by a pre- or post-request script are passed into the next request in the same collection run. Both `hopp.env.set(...)` and legacy `pw.env.set(...)` work, and later requests can reference the value as `<<name>>`.

The update is run-local and does not rewrite environment.json. If it still fails on the latest CLI, please share `hopp --version` and a minimal sanitised collection.
